string name
int8 number_of_agents
---